cogs5e/funcs/lookupFuncs.py

```diff
@@ -25,31 +26,98 @@
 
 class Compendium:
     def __init__(self):
-        self.cfeats = self.load_json('srd-classfeats.json', [])
-        self.classes = self.load_json('srd-classes.json', [])
-        self.conditions = self.load_json('conditions.json', [])
-        self.feats = self.load_json('srd-feats.json', [])
-        self.itemprops = self.load_json('itemprops.json', {})
-        self.monsters = self.load_json('srd-bestiary.json', [])
-        self.names = self.load_json('names.json', [])
-        self.rules = self.load_json('rules.json', [])
-
-        self.spells = [Spell.from_data(r) for r in self.load_json('srd-spells.json', [])]
-        self.backgrounds = [Background.from_data(b) for b in self.load_json('srd-backgrounds.json', [])]
-        self.items = [i for i in self.load_json('srd-items.json', []) if i.get('type') is not '$']
+        self.backgrounds = []
+        self.cfeats = []
+        self.classes = []
+        self.conditions = []
+        self.fancyraces = []
+        self.feats = []
+        self.itemprops = {}
+        self.items = []
+        self.monster_mash = []
+        self.monsters = []
+        self.names = []
+        self.rfeats = []
+        self.rules = []
+        self.spells = []
+        self.srd_backgrounds = []
+        self.srd_items = []
+        self.srd_races = []
+        self.srd_spells = []
+        self.subclasses = []
+
+    async def reload_task(self, mdb=None):
+        wait_for = int(os.getenv('RELOAD_INTERVAL', '300'))  # TODO: decide if 5 minutes is a reasonable default
+        if wait_for > 0:
+            log.info("Reloading data every {} seconds", wait_for)
+            while True:
+                await self.reload(mdb)
+                await asyncio.sleep(wait_for)
+
+    async def reload(self, mdb=None):
+        if mdb is None:
+            await self.load_all_json()
+        else:
+            await self.load_all_mongodb(mdb)
+
+    async def load_all_json(self):
+        self.cfeats = self.read_json('srd-classfeats.json', [])
+        self.classes = self.read_json('srd-classes.json', [])
+        self.conditions = self.read_json('conditions.json', [])
+        self.feats = self.read_json('srd-feats.json', [])
+        self.monsters = self.read_json('srd-bestiary.json', [])
+        self.names = self.read_json('names.json', [])
+        self.rules = self.read_json('rules.json', [])
+        self.srd_backgrounds = self.read_json('srd-backgrounds.json', [])
+        self.srd_items = self.read_json('srd-items.json', [])
+        self.srd_races = self.read_json('srd-races.json', [])
+        self.srd_spells = self.read_json('srd-spells.json', [])
+
+        # Dictionary!
+        self.itemprops = self.read_json('itemprops.json', {})
+
+        self.load_common()
+
+    async def load_all_mongodb(self, mdb):
+        lookup = {d.key: d.object for d in await mdb.static_data.find({}).to_list(length=None)}
```

Comment:
This should probably be `d['key']: d['object']`

Reply:
Yup, found that out when I actually imported some data. Oops.